### ISSUE TITLE
Add more granularity to the ejection of process errors

### ISSUE THREAD

**Issue by espenfl** [aiidateam/aiida_core#2891]
Upon running workchains, we typically get things like this:

`exc_info = (<type 'exceptions.AttributeError'>, AttributeError("'int' object has no attribute 'status'",), None)`

`E AttributeError: 'int' object has no attribute 'status'`

Does not really matter what the error is (this is just an example). It would be great if it is possible to add more details to these ejections. Particularly when running tests on nested workchains it is very hard to know what failed (e.g. which workchain failed for instance). Or maybe there are ways to already do this that I am not seeing. If so I would really appreciate hints on how to proceed.

**Comment by sphuber** [comment]
That cannot be the full stack trace. Can you share the full one please, making sure to get the one straight before it, if it is the tornado future wrapped one

**Comment by espenfl** [comment]
Indeed it is not. Here is a sample (of another error, which is easier to track down due to the explicit name):
```
>       results = run(workchain, **inputs)

aiida_vasp/workchains/tests/test_relax_wc.py:67: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../aiida_core/aiida/engine/launch.py:36: in run
    return runner.run(process, *args, **inputs)
../aiida_core/aiida/engine/runners.py:223: in run
    result, _ = self._run(process, *args, **inputs)
../aiida_core/aiida/engine/runners.py:211: in _run
    process.execute()
../../../env/aiida/lib/python2.7/site-packages/plumpy/processes.py:88: in func_wrapper
    return func(self, *args, **kwargs)
../../../env/aiida/lib/python2.7/site-packages/plumpy/processes.py:1063: in execute
    return self.future().result()
../../../env/aiida/lib/python2.7/site-packages/plumpy/futures.py:34: in result
    return super(Future, self).result(timeout)
../../../env/aiida/lib/python2.7/site-packages/tornado/concurrent.py:238: in result
    raise_exc_info(self._exc_info)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

exc_info = (<class 'aiida.common.exceptions.NotExistent'>, NotExistent('no neighbor with the label output_structure found',), None)

>   ???
E   NotExistent: no neighbor with the label output_structure found

<string>:3: NotExistent
```

Nothing before it. This is it.

**Comment by sphuber** [comment]
This is due to a bug in `plumpy`. I have [opened an issue](https://github.com/aiidateam/plumpy/issues/111) and will be pushing a fix soon

**Comment by sphuber** [comment]
This will be fixed once `plumpy==0.14.1` is released
